hikari/files.py

```diff
@@ -473,6 +473,47 @@ async def read(
         async with self.stream(executor=executor) as reader:
             return await reader.read()
 
+    async def save(
+        self,
+        *,
+        path: typing.Optional[Pathish] = None,
+        executor: typing.Optional[concurrent.futures.Executor] = None,
+        force: bool = False,
+    ) -> None:
+        """Save this resource to disk.
+
+        This writes the resource file in chunks, and so does not load
+        the entire resource into memory.
+
+        Parameters
+        ----------
+        path : typing.Optional[Pathish]
+            The path to save this resource to. If this is a string, the
+            path will be relative to the current working directory. If
+            `builtins.None`, the resource will be saved as
+            `self.filename` in the current working directory.
+        executor : typing.Optional[concurrent.futures.Executor]
+            The executor to run in for blocking operations.
+            If `builtins.None`, then the default executor is used for
+            the current event loop.
+        """
+        loop = asyncio.get_running_loop()
+        path = path or self.filename
+
+        if not isinstance(path, pathlib.Path):
+            path = pathlib.Path(path)
+
+        if path.is_dir():
+            path = path / self.filename
+
+        if path.exists() and not force:
+            raise FileExistsError("file already exists; use `force=True` to overwrite")
```

Comment:
These are I/O blocking.

Instead, you can extract this logic to and run it all in an executor:

```py
def _open_file(path: pathlib.Path, filename: str, force: bool):
    if path.is_dir():
        path = path / self.filename

    if path.exists() and not force:
        raise FileExistsError(f"file {path} already exists; use `force=True` to overwrite")

    return path.open("wb")

# .....

with await loop.run_in_executor(executor, _open_file, path, self.filename, force):
   ...
```